Tools/ardupilotwaf/boards.py

```diff
@@ -771,6 +771,23 @@ def configure_env(self, cfg, env):
                 '-m32',
             ]
 
+        # whitelist of compilers which we should build with -Werror
+        gcc_whitelist = frozenset([
+            ('11','3','0'),
+        ])
+
+        if cfg.env.CC_VERSION in gcc_whitelist or cfg.options.Werror:
```

Comment:
We only should do the whitelist if the selected compiler is gcc.

Reply:
Added this logic and simplified the code in 1783b8f94d1aceb7e2e04c3eea48465511be2819